calc.js

```diff
@@ -0,0 +1,126 @@
+const numberButtons = document.querySelectorAll('[data-number]');
+const mathOperatorButtons = document.querySelectorAll('[data-operation]');
+const allClearButton = document.querySelector('[data-all-clear]');
+const deleteButton = document.querySelector('[data-delete]');
+const equalsButton = document.querySelector('[data-equals]');
+const prevOutput = document.querySelector('[data-prev-operand]');
+const curOutput = document.querySelector('[data-cur-operand]');
+prevOutput.innerText = '';
+curOutput.innerText = '';
+let newPrevOutput = '';
+let newCurOutput = '0';
+let mathOperator = '';
+
+function updateDisplay() {
+  prevOutput.innerText = newPrevOutput;
+  curOutput.innerText = newCurOutput;
+}
+updateDisplay();
+
+function calculate() {
+  let result;
+  newPrevOutput = parseFloat(newPrevOutput);
+  newCurOutput = parseFloat(newCurOutput);
+  if (Number.isNaN(newPrevOutput) || Number.isNaN(newCurOutput)) return;
+  if (mathOperator === '/') {
+    if (newCurOutput === 0) {
+      result = 'Math Err!';
+    } else {
+      result = newPrevOutput / newCurOutput;
+    }
+  } else if (mathOperator === '*') {
+    result = newPrevOutput * newCurOutput;
+  } else if (mathOperator === '+') {
+    result = newPrevOutput + newCurOutput;
+  } else if (mathOperator === '-') {
+    result = newPrevOutput - newCurOutput;
+  } else {
+    return;
+  }
+  newCurOutput = result;
+  mathOperator = undefined;
+  newPrevOutput = '';
+}
+
+function appendNumber(number) {
+  if (newCurOutput === '0' && number === '0') return;
+  if (newCurOutput.includes('.') && number === '.') return;
+  if (newCurOutput === '0' && number >= '1' && number <= '9') {
+    newCurOutput = ''.toString() + number.toString();
```

Comment:
- [ ] Considering that an empty string is a string, using the `toString` method seems unnecessary here. You could just reassign the `newCurOutput` to `number.toString()` to achieve the same result for this and similar cases

Reply:
Thanks... I'd fix that immediately!